packages/iris-grid/src/sidebar/conditional-formatting/ConditionEditor.tsx

```diff
@@ -0,0 +1,254 @@
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import Log from '@deephaven/log';
+import { TableUtils } from '../..';
+import {
+  StringCondition,
+  DateCondition,
+  getLabelForNumberCondition,
+  getLabelForDateCondition,
+  getLabelForStringCondition,
+  NumberCondition,
+  ModelColumn,
+} from './ConditionalFormattingUtils';
+
+const log = Log.module('ConditionEditor');
+
+export interface ConditionConfig {
+  condition: NumberCondition | StringCondition | DateCondition;
+  value?: string | number;
+  start?: number;
+  end?: number;
+}
+
+export interface ConditionEditorProps {
+  column: ModelColumn;
+  config: ConditionConfig;
+  onChange?: (config: ConditionConfig, isValid: boolean) => void;
+}
+
+const DEFAULT_CALLBACK = () => undefined;
+
+const numberConditionOptions = [
+  NumberCondition.IS_EQUAL,
+  NumberCondition.IS_NOT_EQUAL,
+  NumberCondition.IS_BETWEEN,
+  NumberCondition.GREATER_THAN,
+  NumberCondition.GREATER_THAN_OR_EQUAL,
+  NumberCondition.LESS_THAN,
+  NumberCondition.LESS_THAN_OR_EQUAL,
+].map(option => (
+  <option key={option} value={option}>
+    {getLabelForNumberCondition(option)}
+  </option>
+));
+
+const stringConditions = [
+  StringCondition.IS_EXACTLY,
+  StringCondition.IS_NOT_EXACTLY,
+  StringCondition.CONTAINS,
+  StringCondition.DOES_NOT_CONTAIN,
+  StringCondition.STARTS_WITH,
+  StringCondition.ENDS_WITH,
+].map(option => (
+  <option key={option} value={option}>
+    {getLabelForStringCondition(option)}
+  </option>
+));
+
+const dateConditions = [
+  DateCondition.IS_EXACTLY,
+  DateCondition.IS_NOT_EXACTLY,
+  DateCondition.IS_BEFORE,
+  DateCondition.IS_BEFORE_OR_EQUAL,
+  DateCondition.IS_AFTER,
+  DateCondition.IS_AFTER_OR_EQUAL,
+].map(option => (
+  <option key={option} value={option}>
+    {getLabelForDateCondition(option)}
+  </option>
+));
+
+const ConditionEditor = (props: ConditionEditorProps): JSX.Element => {
```

Comment:
Getting an error thrown and the Value box disappears in Condition Editor: 
```
[ConditionEditor] Unable to create formatting rule. Invalid value undefined
console.<computed> @ index.js:1
```

Steps to Reproduce:
1) Create miami table
```
miami = newTable(
    stringCol("Month","Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"),
    intCol("Temp", 60, 62, 65, 68, 73, 76, 77, 77, 76, 74, 68, 63),
    doubleCol("Rain", 1.62, 2.25, 3.00, 3.14, 5.34, 9.67, 6.50, 8.88, 9.86, 6.33, 3.27, 2.04)
)
```
2) Create Row format for Month == 'Apr'
3) Open Conditional Formatting editor, go to Rows, and change column to Temp. The `Values` input field disappears:
![image](https://user-images.githubusercontent.com/4505624/152348127-494bcba8-02d6-4b61-8488-96dcaaa830fc.png)

Reply:
Fixed